flycheck.el

```diff
@@ -312,6 +312,22 @@ respectively."
   :safe #'flycheck-symbol-list-p)
 (make-variable-buffer-local 'flycheck-disabled-checkers)
 
+(defvar-local flycheck-automatically-disabled-checkers nil
+  "List of syntax checkers automatically disabled for this buffer.
+
+A checker can be automatically disabled in two cases:
+
+1. Its `:enabled' predicate returned false.
+2. It returned too many errors (see `flycheck-checker-error-threshold').
+
+The command `flycheck-verify-setup' will clear this variable and
+reverify all checkers for the current buffer.  Checkers that fail
+the `:enabled` test will remain automatically disabled.  Checkers
+that pass will be enabled.
```

Comment:
I think this is a bit dangerous: don't we fear that we'll have heisenbugs were something doesn't work, we ask for the contents of `verify`, and and the bugs go away?

Reply:
That is risk indeed.  This convenience may in fact mystify users.  Better to have the user trigger the reverification (using buttons like in #1478).

Besides, the reverification can delay opening the verify-setup buffer, as in js-mode where the enabled test for `eslint` takes its sweet time.

I'll remove this auto-reset and we'll deal with the interaction bit in #1478.